openwrt-openwisp-monitoring/files/monitoring.agent

```diff
@@ -267,6 +286,11 @@ main() {
 		MONITORED_INTERFACES=$(echo "$MONITORED_INTERFACES" | tr -d '"')
 		save_data
 	elif [ "$MODE" = "send" ]; then
+		if [ -z "$UUID" ] || [ -z "$KEY" ]; then
```

Comment:
I see redundancy again here, because the `register` function already checks whether UUID and key are present, I would do the following:

- remove the if here
- rename `register` to `wait_until_registered`
- change the `wait_until_registered` to check whether UUID and KEY are present first, then only if not present look into /etc/config/openwisp and finally if uuid and key are present there export the values or return 1

Reply:
Updated accordingly.